Source/Diagnostics/ParticleIO.cpp

```diff
@@ -299,3 +300,113 @@ storePhiOnParticles ( PinnedMemoryParticleContainer& tmp,
+    );
+
+    constexpr auto lev0=0;
+
+    amrex::MultiFab const& Ex = *warpx.m_fields.get(FieldType::Efield_fp, Dir{0}, lev0);
+    amrex::MultiFab const& Ey = *warpx.m_fields.get(FieldType::Efield_fp, Dir{1}, lev0);
+    amrex::MultiFab const& Ez = *warpx.m_fields.get(FieldType::Efield_fp, Dir{2}, lev0);
+    amrex::MultiFab const& Bx = *warpx.m_fields.get(FieldType::Bfield_fp, Dir{0}, lev0);
+    amrex::MultiFab const& By = *warpx.m_fields.get(FieldType::Bfield_fp, Dir{1}, lev0);
+    amrex::MultiFab const& Bz = *warpx.m_fields.get(FieldType::Bfield_fp, Dir{2}, lev0);
+
+    const amrex::XDim3 dinv = WarpX::InvCellSize(lev0);
+    const bool galerkin_interpolation = WarpX::galerkin_interpolation;
+    const int nox = WarpX::nox;
+    const int n_rz_azimuthal_modes = WarpX::n_rz_azimuthal_modes;
+
+#ifdef AMREX_USE_OMP
+#pragma omp parallel if (amrex::Gpu::notInLaunchRegion())
+#endif
+    for (PinnedParIter pti(tmp, lev0); pti.isValid(); ++pti) {
+
+        const auto Ex_grid = Ex[pti].array();
+        const auto Ey_grid = Ey[pti].array();
+        const auto Ez_grid = Ez[pti].array();
+        const auto Bx_grid = Bx[pti].array();
+        const auto By_grid = By[pti].array();
+        const auto Bz_grid = Bz[pti].array();
+
+        const auto ex_type = Ex.ixType();
+        const auto ey_type = Ey.ixType();
+        const auto ez_type = Ez.ixType();
+        const auto bx_type = Bx.ixType();
+        const auto by_type = By.ixType();
+        const auto bz_type = Bz.ixType();
+
+        const auto getPosition = GetParticlePosition<PIdx>(pti);
+        amrex::ParticleReal* Ex_particle_arr = pti.GetStructOfArrays().GetRealData(Ex_index).dataPtr();
+        amrex::ParticleReal* Ey_particle_arr = pti.GetStructOfArrays().GetRealData(Ey_index).dataPtr();
+        amrex::ParticleReal* Ez_particle_arr = pti.GetStructOfArrays().GetRealData(Ez_index).dataPtr();
+        amrex::ParticleReal* Bx_particle_arr = pti.GetStructOfArrays().GetRealData(Bx_index).dataPtr();
+        amrex::ParticleReal* By_particle_arr = pti.GetStructOfArrays().GetRealData(By_index).dataPtr();
+        amrex::ParticleReal* Bz_particle_arr = pti.GetStructOfArrays().GetRealData(Bz_index).dataPtr();
+
+        const auto box = pti.tilebox();
+        const amrex::XDim3 xyzmin = WarpX::LowerCorner(box, lev0, 0._rt);
+        const Dim3 lo = lbound(box);
+
+        // Loop over the particles and compute the EM field using the doGatherShapeN function
+        amrex::ParallelFor( pti.numParticles(),
+            [=] AMREX_GPU_DEVICE (long ip) {
+               amrex::ParticleReal xp, yp, zp;
+               getPosition(ip, xp, yp, zp);
+
+                Ex_particle_arr[ip] = 0._rt;
+                Ey_particle_arr[ip] = 0._rt;
+                Ez_particle_arr[ip] = 0._rt;
+                Bx_particle_arr[ip] = 0._rt;
+                By_particle_arr[ip] = 0._rt;
+                Bz_particle_arr[ip] = 0._rt;
```

Comment:
Could we add the external fields as part of this call?

Reply:
See for instance: https://github.com/ECP-WarpX/WarpX/blob/development/Source/Particles/ElementaryProcess/Ionization.H#L115